### PR TITLE
fix: Plugins without sidecars were not injected

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
At the beginning, theia component was part of the devWorkspace and then we
had access to the atttributes from the DevWorkspace object
Now, it's part of the devWorkspaceTemplate object

Switch to the flattened devfile instead so nothing to query externally
Also, in order to reduce sed usage, use jq/yq instead as there was also a bug
on empty list

Added some unit tests as well

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20718

### How to test this PR?
Start the workspace from the issue
Eslint should not be there (check output widget inside theia or command palette)
if you check the logs of the init container you might as well see tons of errors

Open a terminal and drop anything located inside /plugins folder

Edit the DevWorkspaceTemplate object

Search for the `remote-runtime` component
```yaml
app.kubernetes.io/component: remote-runtime-injector
app.kubernetes.io/part-of: che-theia.eclipse.org
```

and replace the default image by `quay.io/fbenoit/che-theia-endpoint-runtime-binary:20211104c`

restart the workspace
eslint should be there

Also you might check the log trace of the init-container and see something like
```
downloading https://download.jboss.org/jbosstools/vscode/3rdparty/vscode-eslint/vscode-eslint-2.1.1-1e15d3.vsix to /plugins for component theia
```

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)
- [x] Optional Companion PR for updating the HappyPath tests is approved and ready to be merged

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=next

Change-Id: Ic8b417484596f8440a157e0d24cd7bbf4da9c40c
Signed-off-by: Florent Benoit <fbenoit@redhat.com>

### CODE REVIEW

File: dockerfiles/theia-endpoint-runtime-binary/src/entrypoint.sh [hunk 1]
```diff
@@ -11,48 +11,95 @@
...
+}
+
+# extract the destination folder from one item of the analyze
+# parameter: $1 is the component data item
+extract_dest_from_item() {
+  # also remove the local-dir:// prefix
+  jq -n "$1" | jq -r '.[].env[] | select(.name == "THEIA_PLUGINS") | .value' | sed 's|local-dir://\([^"][^"]*\)|\1|'
+}
+
+# extract the destination folder from one item of the analyze
```
Comment (svor) [inline dockerfiles/theia-endpoint-runtime-binary/src/entrypoint.sh:34]: should be URL instead of destination

**Review by svor — COMMENTED** [review]
Works with the sample described in the issue:

screenshot-che-eclipse-che apps cluster-pcccs pcccs sandbox279 opentlc com-2021 11 04-13_01_18

Also tried with scala sample: https://github.com/svor/scala-sbt/tree/devfile2 (contains 2 extensions, one of them without sidecar):
screenshot-che-eclipse-che apps cluster-pcccs pcccs sandbox279 opentlc com-2021 11 04-13_22_55

Works as expected!

File: dockerfiles/theia-endpoint-runtime-binary/src/entrypoint.sh (hunk 1 above)
Comment (benoitf) [reply dockerfiles/theia-endpoint-runtime-binary/src/entrypoint.sh]: yes, thanks
fixed

**Comment by benoitf** [comment]
[crw-ci-test --rebuild]

**Review by svor — APPROVED** [review]

**Comment by benoitf** [comment]
[crw-ci-test --rebuild]

**Comment by benoitf** [comment]
[crw-ci-test --rebuild]

**Comment by benoitf** [comment]
[crw-ci-test --rebuild]

**Review by azatsarynnyy — APPROVED** [review]

**Pull request merged**